driver/src/price_finding/linear_optimization_price_finder.rs

```diff
@@ -104,68 +171,33 @@ fn parse_token(key: &str) -> Result<u16, PriceFindingError> {
     ))
 }
 
-fn parse_price_value(value: &Value) -> Result<u128, PriceFindingError> {
-    match value {
-        Value::String(value) => value.parse::<u128>().map_err(|err| {
-            PriceFindingError::new(
-                format!("Failed to parse price string: {}", err).as_ref(),
-                ErrorKind::ParseIntError,
-            )
-        }),
-        Value::Number(value) => {
-            // Note there is no direct conversion from Number to u128
-            u128::deserialize(value).map_err(|err| {
-                PriceFindingError::new(
-                    format!("Failed to deserialize JSON Number: {}", err).as_ref(),
-                    ErrorKind::JsonError,
-                )
-            })
-        }
-        Value::Null => Ok(0u128),
-        other => Err(PriceFindingError::new(
-            format!("Expected numerical string but got '{}'", other).as_ref(),
-            ErrorKind::JsonError,
-        )),
-    }
+fn parse_price(price: &Option<String>) -> Result<u128, PriceFindingError> {
+    price.as_ref().map_or(Ok(0), |price| {
```

Comment:
Actually I was hoping we could actually leave zero prices out of the price map and adjust the logic in a couple other places to infer zero as a default when the prices are not included.

Reply:
We would then no longer need the "TokenNotFound" error in objective function evaluation.

Reply:
At this point I wanted to do everything as before - just with serde. Lets change the  behavior for missing prices in a separate PR.

Reply:
Yes, that is totally fine, but maybe we can create a new issue out of this.